spec/updater_spec.rb

```diff
@@ -7,77 +7,84 @@
+      expect(output.string).to include(Constants::NO_CONTACTS_MESSAGE)
     end
 
-    it 'asks the user which field it wants to update and the value for it' do
+    it 'does not ask updating prompt if database is empty' do
+      user_interface = UserInterface.new(StringIO.new("\n"), output, Validator.new)
+      database = ArrayDatabase.new
       updater = described_class.new(user_interface, database)
 
       updater.run
 
+      expect(output.string).to_not include(Constants::CONTACT_INDEX_PROMPT)
+    end
+
+    it 'asks the user for a contact to update' do
+      run_setup_with_input(quick_exit_responces)
+
+      expect(output.string).to include(Constants::CONTACT_INDEX_PROMPT)
+    end
+
+    it 'asks the user which field it wants to update and the value for it' do
+      run_setup_with_input(quick_exit_responces)
+
       expect(output.string).to include(Constants::FIELD_CHOICE_PROMPT, Constants::FIELDS_TO_PROMPTS[:name])
     end
 
     it 'displays the contact with the data updated' do
-      updater = described_class.new(user_interface, database)
-
-      updater.run
+      run_setup_with_input(quick_exit_responces)
 
       expect(output.string).to include("Name:    irrelevant\nAddress: Some address")
     end
 
     it 'asks the user if they want to edit another field' do
-      updater = described_class.new(user_interface, database)
-
-      updater.run
+      run_setup_with_input(quick_exit_responces)
 
       expect(output.string).to include(Constants::ANOTHER_EDIT_PROMPT)
     end
 
     it 'collects another field and value if they want to edit another field' do
-      input = StringIO.new("0\nname\nirrelevant\ny\naddress\nirrelevant\nn\nn\n")
-      user_interface = UserInterface.new(input, output, validator)
-      updater = described_class.new(user_interface, database)
-
-      updater.run
+      input = "0\nname\nirrelevant\ny\naddress\nirrelevant\nn\nn\n"
+      run_setup_with_input(input)
 
       expect(output.string.scan(Constants::FIELD_CHOICE_PROMPT).length).to eq(2)
     end
 
     it 'asks if they want to edit another contact if they do not want to edit another field' do
-      updater = described_class.new(user_interface, database)
-
-      updater.run
+      run_setup_with_input(quick_exit_responces)
 
       expect(output.string).to include(Constants::ANOTHER_UPDATE_PROMPT)
     end
 
     it 'asks for another contact index if they want to edit another contact' do
-      input = StringIO.new("0\nname\nirrelevant\ny\naddress\nirrelevant\nn\ny\n" + quick_exit_responces)
-      user_interface = UserInterface.new(input, output, validator)
-      updater = described_class.new(user_interface, database)
-
-      updater.run
+      input = "0\nname\nirrelevant\ny\naddress\nirrelevant\nn\ny\n" + quick_exit_responces
+      run_setup_with_input(input)
 
       expect(output.string.scan(Constants::CONTACT_INDEX_PROMPT).length).to eq(2)
     end
   end
 
+  def run_setup_with_input(string)
```

Comment:
you are not running a setup though, you are running the updater. This tests are to test the updater.

```suggestion
  def run_setup_with_input(string)
  def run_updater_with_input(string)
```

Reply:
I saw it as, I'm running a setup (of the updater).

Reply:
really, I'm trying to expain that I didn't just copy your code, I put it in my way and refactored it till it was the same as yours. Also your naming convention made sense as it is a method that sets up.

Reply:
yeah, that's fine to not copy my code exactly, 
it's just that in this case the naming could be improved, as "run setup" can have other meanings